packages/core/src/lib/utils/apr.ts

```diff
@@ -0,0 +1,132 @@
+
+    var tokens: number = 0
+    for (var release of sortedReleaseSchedule) {
+        // Skip all schedules that ended before today.
+        if (release.endDate.isBefore(startDate)) {
+            continue;
+        }
+        // If the start date is after end date, we have come to the end of all releases we should consider.
+        if (release.startDate.isAfter(endDate)) {
+            break;
+        }
+        // All releases from here are part of the window.
+        // The case where this release started before today.
+        if (release.startDate.isBefore(startDate)) {
+
+            // Need to deduct 1 day from endDate to make it an inclusive end date.
+            let earlierInclusiveEndDate = moment.min(endDate.clone().subtract(1, "days"), release.endDate);
+
+            // Number of days left in this release.
+            let daysLeft: number = calculateDaysInclusive(startDate, earlierInclusiveEndDate);
+            let totalPeriod: number = calculateDaysInclusive(release.startDate, release.endDate);
+            tokens += (daysLeft / totalPeriod) * release.tokenReleaseAmount;
+        }
+
+        // The case where this release ends after our search window.
+        else if (release.endDate.isAfter(endDate)) {
+            let daysLeft: number  = Math.round(endDate.diff(release.startDate, 'days', true));
+            let totalPeriod: number  = calculateDaysInclusive(release.startDate, release.endDate);
+            tokens += (daysLeft / totalPeriod) * release.tokenReleaseAmount;
+        }
+
+        // In the final case, the entire period falls within our window.
+        else {
+            tokens += release.tokenReleaseAmount;
+        }
+    }
+
+    return tokens;
+}
+
+// Converts the releaseSchedule into ReleaseSchedule[] and sorts it by start date.
+function getSortedReleaseSchedule(releaseSchedule: ScheduledTokenReleaseSDKType[]) {
+        let releaseScheduleTimes = releaseSchedule.map((schedule) => {
+            return CreateReleaseSchedule(schedule.start_date, schedule.end_date, schedule.token_release_amount);
+        })
+
+        // Sort release schedule in increasing order of start time.
+        let sortedReleaseSchedule = releaseScheduleTimes.sort((x, y) => {
+            if (x.startDate.isAfter(y.startDate)) {
+                return 1;
+            }
+            else if (y.startDate.isAfter(x.startDate)) {
+                return -1;
+            }
+            return 0;
+        })
+
+        return sortedReleaseSchedule;
+}
+
+// Returns the number of days in the window inclusive of the start and end date.
+function calculateDaysInclusive(startDate: Moment, endDate: Moment) {
+    return Math.round(endDate.diff(startDate, 'days', true)) + 1;
+}
+
+interface ReleaseSchedule {
+    startDate: Moment;
+    endDate: Moment;
+    tokenReleaseAmount: number;
+}
+
+function CreateReleaseSchedule(start_date: string, end_date: string, token_release_amount: Long): ReleaseSchedule {
```

Comment:
can you make function names camelcase? so `createReleaseSchedule`

Reply:
Fixed thanks!